htsget.md

```diff
@@ -443,6 +445,122 @@ Initial guidelines, which we expect to revise in light of future experience:
+
+<table>
+
+<tr markdown="block"><td>
+
+`datatype`  
+_optional string_
+</td><td>
+Indicates the htsget datatype category ('reads' or 'variants') served by the ticket endpoint related to this service-info endpoint. Use either:
+
+* reads
+* variants
+</td></tr>
+<tr markdown="block"><td>
+
+`formats`  
+_optional array of strings_
+</td><td>
+
+List of requested `format` that can be satisfied. Allowed values:
+
+* reads: BAM and/or CRAM
+* variants: VCF and/or BCF
+</td></tr>
+<tr markdown="block"><td>
+
+`fieldsParameterEffective`  
+_optional boolean_
+</td><td>
+
+Indicates whether the service is capable of returning only a subset of available fields based on the `fields` query parameter.
+</td></tr>
+<tr markdown="block"><td>
+
+`tagsParametersEffective`  
+_optional boolean_  
+</td><td>
+
+Indicates whether the service is capable of server-side result tag inclusion/exclusion using the `tags` and `notags` query parameters.
+</td></tr>
+</table>
+
+Example service-info response:
+
+```
+{ 
+   "id":            "net.exampleco.htsget",
+   "name":          "ExampleCo genomics data service",
+   "version":       "0.1.0",
+   "organization":  {
+      "name":       "ExampleCo",
+      "url":        "https://exampleco.com"
+   },
+   "type":  {
+      "group":        "org.ga4gh",
+      "artifact":     "htsget",
+      "version":      "1.3.0"
+   },
+   "htsget": {
+      "datatype": "reads",
+      "formats":  ["BAM", "CRAM"],
+      "fieldsParameterEffective": true,
+      "tagsParametersEffective": false
+   }
+}
+```
+
+# GA4GH Service Registry
+
+The [GA4GH Service Registry API specification](https://github.com/ga4gh-discovery/ga4gh-service-registry) allows information about GA4GH-compliant web services, including htsget services, to be aggregated into registries and made available via a standard API. The following considerations SHOULD be followed when registering htsget services within a service registry.
+
+* Endpoints for different htsget data types should be registered as separate entities within the registry. If an htsget service provides both `reads` and `variants` data, both endpoints should be registered.
+* The `url` property should reference the API's `service-info` endpoint for a single data type (i.e. an htsget reads API registration should provide the URL to the reads `service-info` endpoint, an htsget variants API registration should provide the URL to the variants `service-info` endpoint). Clients should be able to assume that by replacing the URL's "service-info" string with an object id, they will hit the corresponding `reads`/`variants` endpoint.
```

Comment:
Are we happy with this assumption? I might be missing something but:

- If we're reserving `/service-info` so that it's not allowed as a sample ID, then it seems it would be fine to leave that off instead of making the client snip it.
- If we're saying we could use this to declare the service info endpoint as something else like `/service-info2` so that the server can still have a sample ID=`service-info`, we should specifically advise that client is supposed to strip whatever the last URL component is, and on the whole it seems unfortunate to make this an htsget-specific caveat)

My opinion is still that the first approach is a simple and basically acceptable way forward. If in future we motivate a change to service registry so that it admits a query string for service-info then I think we could relax the reservation constraint at that time.

(comments welcome from all, tagging @jmarshall @brainstorm @jb-adams)

Reply:
I'm +1 for the former. Registering the base url (ie. without "service-info") is much more in line with the service info spec and the networks team's plans for federating services of mixed type. Requiring that "service-info" be reserved for this endpoint is simple and will lead to better interoperability.

Reply:
I totally second Jeremy.